engine/script/src/script.cpp

```diff
@@ -139,6 +139,8 @@ namespace dmScript
         lua_getglobal(L, RANDOM_SEED);
         uint32_t* seed = (uint32_t*) lua_touserdata(L, -1);
         *seed = luaL_checkint(L, 1);
+        // discard first value to avoid repeated values
```

Comment:
This is what's done in Lua 5.2 for instance:

https://github.com/lua/lua/blob/v5-2/lmathlib.c#L228

Reply:
Is this needed after we set the seed in `math.randomseed(seed)`?

Reply:
I'm not sure I understand?

Reply:
I'm not sure either. I really thought this code was inside the "math.rand()" fuction 🤔 

But still, the question I intended is still valid.
We have two use cases:

1) call math.rand()  
2) call math.randomseed() and then math.rand()

This PR only pre-warms the 2) case, inside the math.randomseed().
So, if you're using 1), you'll need to do "math.rand()" yourself in your script?

I.e. do we need a dmMath::Rand(seed); inside the [dmScript::Initialize()](https://github.com/defold/defold/blob/dev/engine/script/src/script.cpp#L179) as well?

Reply:
Ah, ok, I understand. I think it is much less of a problem when you call math.random() without first setting a seed. But yeah, maybe it's best to also call dmMath::Rand(seed) once during init.

Reply:
Not sure how it's less of a problem?

The seed is just an external integer, which is then passed to the dmMath::Rand() function.
The Rand function doesn't care if the seed is 0 or not.

So, if seed!=0 is a problem, then I assume seed=0 is a problem too.

Reply:
I think my point is supported by this issue (i.e. the default seed=0 is low, thus will generate issues):
https://github.com/defold/defold/issues/3753

Also, is calling it +1 times fixing the observed behavior in #3753?

Reply:
Yes, calling it once is sufficient according to my own tests.